logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/conversation/message/ApplicationMessageHandler.kt

```diff
@@ -111,47 +100,71 @@ internal class ApplicationMessageHandlerImpl(
         // The message was previously received with just metadata info, so let's update it with the raw data info
         return persistedMessage.copy(
             content = assetMessageContent.copy(
                 value = assetMessageContent.value.copy(
                     remoteData = remoteData
                 )
-            ),
-            visibility = Message.Visibility.VISIBLE
+            ), visibility = Message.Visibility.VISIBLE
         )
     }
 
     private suspend fun isSenderVerified(messageId: String, conversationId: ConversationId, senderUserId: UserId): Boolean {
         var verified = false
         messageRepository.getMessageById(
-            messageUuid = messageId,
-            conversationId = conversationId
+            messageUuid = messageId, conversationId = conversationId
         ).onSuccess {
             verified = senderUserId == it.senderUserId
         }
         return verified
     }
 
-    private suspend fun processSignaling(senderUserId: UserId, signaling: MessageContent.Signaling) {
-        when (signaling) {
+    private suspend fun processSignaling(signaling: Message.Signaling) {
+        when (val content = signaling.content) {
             MessageContent.Ignored -> {
-                logger
-                    .i(message = "Ignored Signaling Message received: $signaling")
+                logger.i(message = "Ignored Signaling Message received: $signaling")
             }
 
             is MessageContent.Availability -> {
-                logger
-                    .i(message = "Availability status update received: ${signaling.status}")
-                userRepository.updateOtherUserAvailabilityStatus(senderUserId, signaling.status)
+                logger.i(message = "Availability status update received: ${content.status}")
+                userRepository.updateOtherUserAvailabilityStatus(signaling.senderUserId, content.status)
+            }
+
+            is MessageContent.Reaction -> persistReaction(content, signaling.conversationId, signaling.senderUserId, signaling.date)
+            is MessageContent.DeleteMessage -> handleDeleteMessage(content, signaling.conversationId, signaling.senderUserId)
+            is MessageContent.DeleteForMe -> deleteForMeHandler.handle(content, signaling.conversationId)
+            is MessageContent.Calling -> {
+                logger.d("MessageContent.Calling")
+                callManagerImpl.value.onCallingMessageReceived(
+                    message = signaling,
+                    content = content,
+                )
             }
+
+            is MessageContent.TextEdited -> editTextHandler.handle(signaling, content)
+            is MessageContent.LastRead -> lastReadContentHandler.handle(signaling, content)
+            is MessageContent.Cleared -> clearConversationContentHandler.handle(signaling, content)
+            is MessageContent.Receipt -> TODO("Receiving of read receipts not yet implemented")
```

Comment:
I'm thinking if this will cause a crash if immediately merged to develop ? Or is it part of a set of PR's 🤔

Reply:
Not yet, as the mapping from Protobuf to Receipt is not done yet, so there's a total of 0 instantiations of `MessageContent.Receipt`.

This just slipped through as it was when creating this class that I noticed that we could improve some stuff :D